kinley-shayne/server.js

```diff
@@ -0,0 +1,199 @@
+'use strict';
+
+const pg = require('pg');
+const fs = require('fs');
+const express = require('express');
+const bodyParser = require('body-parser');
+const requestProxy = require('express-request-proxy');
+const PORT = process.env.PORT || 3000;
+const app = express();
+// const conString = 'postgres://USERNAME:PASSWORD@HOST:PORT';
+const conString = process.env.DATABASE_URL; // DONE: Don't forget to set your own conString
+const client = new pg.Client(conString);
+client.connect();
+client.on('error', err => console.error(err));
+
+app.use(bodyParser.json());
+app.use(bodyParser.urlencoded({extended: true}));
+app.use(express.static('./public'));
+
+app.get('/new', (request, response) => response.sendFile('new.html', {root: './public'}));
+app.get('/admin', (request, response) => response.sendFile('admin.html', {root: './public'}));
+app.get('/articles', (request, response) => {
+  client.query(`
+    SELECT * FROM articles
+    INNER JOIN authors
+      ON articles.author_id=authors.author_id;`
+  )
+  .then(result => response.send(result.rows))
+  .catch(console.error);
+});
+
+// DONE: Whenever a "get" request is sent to a route starting with "/github/", use the remainder of that url to access github, with your token as the Authorization header. You should *only* make direct requests to Github from server.js, not from the front-end. What you send back will be a collection of repositories, as an array of objects.
+
+app.get('/github/*', (request, response) => {
+  (requestProxy(
+    {
+      url: `https://api.github.com/${request.params[0]}`,
+      headers: {
+        Authorization: `token ${process.env.GITHUB_TOKEN}`
+      }
+    }
+  ))(request, response)
+});
```

Comment:
Just so you know, you do not need to wrap this in an anonymous function for requestProxy to work. [express-request-proxy docs](https://www.npmjs.com/package/express-request-proxy#wildcard-routes)